### DISCUSSION TITLE
Cluster Setup With RabbitMQ.Client.dll 3.4.3 For .NET 4.0 Framework

### DISCUSSION THREAD

**Discussion by Pramod34** [rabbitmq/rabbitmq-dotnet-client#1462] [General]
We have an application hosted on .NET 4.0 Framework using RabbitMQ.Client.dll 3.4.3 version. We are creating a cluster environment for our PROD environment setup for High Avaliabilty (HA). But RabbitMQ.Client.dll v3.4.3 doesn't support passing multiple IPAddresses to CreateConnection() method.

The apporach is looping through the IPAddresses and tried is to connect one node (One IPAddress), if connection is successfull we will use that node connection, if connection is unsuccessfull we will try with other node (other IPAddress) in the cluster setup. The problem with this approach is there will be downtime untill it reconnects to other nodes in the cluster.

Can anyone help me out how to pass multiple IPAddresses or achiving HA with RabbitMQ.Client.dll v3.4.3?

I would really appreciate if solution or answer

**Comment by michaelklishin** [comment]
The client/.NET framework compatibility table can be [found in the docs](https://www.rabbitmq.com/dotnet.html).

You can use .NET client versions going back to 2.0 with the latest RabbitMQ versions. Not all features won't be available and there will be subtle behavior changes but very likely all key operations will work just fine.

**Comment by michaelklishin** [comment]
Passing multiple IP addresses to more modern versions of the client does nothing more than trying them in order until connection to one endpoint succeeds. See [Clustering and Clients](https://www.rabbitmq.com/clustering.html#clustering-and-clients).

All messaging protocols RabbitMQ supports use a single connection to one node. [RabbitMQ Stream](https://rabbitmq.com/streams.html) clients connect to multiple nodes at the same time [if allowed to do so](https://blog.rabbitmq.com/posts/2021/07/connecting-to-streams/). But that's a completely different protocol, client library and key data structures underneath. And there isn't and won't be a .NET 4.0-compatible Stream client version.

**Comment by lukebakken** [comment]
@Pramod34 you are using the correct procedure. The current code that takes a list of addresses does not do anything different than what your code does.

You should use a load balancer (like haproxy) between your applications and RabbitMQ to achieve round-robin connections.
